doc/specs/#658 - WinGet Download.md

```diff
@@ -0,0 +1,209 @@
+---
+author: Roy MacLachlan RDMaclachlan/roy.maclachlan@microsoft.com
+created on: 2023-02-09
+last updated: 2023-02-09
+issue id: 658
+---
+
+# Spec Title
+
+"For [#658](https://github.com/microsoft/winget-cli/issues/658)"
+
+## Abstract
+
+This spec describes the functionality and high-level implementation design for the downloading of applications using the Windows Package Manager.
+
+## Inspiration
+
+This is inspired by customer feedback, and a need for broader application deployments:
+* Customers want to share the installer with an offline device. 
+
+## Solution Design
+
+The Windows Package Manager will implement a new command `WinGet Download` that will offer customers with the ability to download a single package's installers. The new command will make use of the following optional parameter values:
+* `--id <Identifier>`
+* `--name <name>`
+* `--moniker <moniker>`
+* `--scope <scope>`
+* `--locale <locale>`
+* `--fileName <name>`
+* `-v, --version <version>`
+* `-s, --source <Source>`
+* `-a, --architecture <architecture>`
+* `-e, --exact`
+* `-o, --output <path>`
+* `-t, --type <installer type>`
+* `--ignore-security-hash`
+* `--accept-package-agreements`
+* `--help`
+* `--wait`
+* `--verbose, --verbose-logs`
+
+An update to the existing schema is required for the inclusion of the `<boolean> isDownloadable` property. If this property is set to `True` or is not specified, then the package's installer will be considered downloadable. If the property is set to `False`, then the request to download the package's installer will be refused.
```

Comment:
Why is the `isDownloadable` needed? Users need to be able to download packages in the install process, so to specify that they can download it and run it, but can’t download it and not run it doesn’t make a lot of sense to me, especially since users could just run the install flow and terminate the thread between download and install to bypass this

Reply:
> Why is the `isDownloadable` needed? Users need to be able to download packages in the install process, so to specify that they can download it and run it, but can’t download it and not run it doesn’t make a lot of sense to me, especially since users could just run the install flow and terminate the thread between download and install to bypass this

This is originally from msstore type where Store needs to limit what apps could be downloadable, apps marked as downloadable needs app developer approval.

Reply:
Thanks for the explanation!